cql3/statements/create_table_statement.cc

```diff
@@ -71,7 +71,7 @@ std::vector<column_definition> create_table_statement::get_columns() const
 }
 
 future<std::tuple<::shared_ptr<cql_transport::event::schema_change>, std::vector<mutation>, cql3::cql_warnings_vec>>
-create_table_statement::prepare_schema_mutations(query_processor& qp, api::timestamp_type ts) const {
+create_table_statement::prepare_schema_mutations(query_processor& qp, const query_options&, api::timestamp_type ts) const {
```

Comment:
Again, your commit message doesn't explain **why** you are doing this change. In this case, you are changing a gazillion places in the code, so I have to wonder how come none of these other statements never needed the "query options" and this one does. After all, each one of these other statements had a lot options on its own and I assume knew how to save them into the statement object and didn't need to pass options into `prepare_schema_mutations` specifically, so why do you need to add it now?

Reply:
I think I finally understood what's going one:

Although we did have various statements that can take bound variables, including SELECT, UPDATE, none of the schema-modification statements (CREATE .., ALTER ...) ever took bound variables, and this is the first time you're introducing this possibility. So, because all these statements inherit from one class "schema_altering_statement" you need to add this possibility to this class and then unfortunately add the new parameter to all of its implementations.

But this raises for me two questions, which would be good to answer here *and* in the commit message:
1. Why, if we never had bound variables in any schema-altering statement, did we suddenly decide to do it in this case? Especially given that (as you said) Cassandra doesn't allow it? Do we think anyone would ever prepare ALTER TABLE statements and want to use bound variables? Or maybe we believe there is simply no reason why it should not work - the "principle of least surprise" means we should allow it?
2. Is it impossible (without extensive change to the code) to add this extra paramater only to the one alter_table_statement class, and not to the schema_altering_statement parent class and all of its subclasses like you did in this patch?

Reply:
1. I think that was the "principle of least surprise". Kostja asked about it and I've added the support. In general, we support `?` in USING TIMESTAMP, so I thought it's a good idea to add it.
2. Schema statement are executed via schema_altering_statement interface, that's why I couldn't add the parameter only to alter table.